agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/CommandExecutor.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.applicationinsights.agent.internal.init;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+
+class CommandExecutor {
+
+  private CommandExecutor() {}
+
+  static String execute(ProcessBuilder processBuilder) {
+    IllegalStateException exitException = null;
+    String result = null;
+    try {
+      Process process = processBuilder.start();
+      int exitValue = process.waitFor();
+      exitException =
+          buildExitValueExceptionIfNecessary(processBuilder.command(), exitValue, process);
+      if (exitException == null) {
+        InputStream inputStream = process.getInputStream();
```

Comment:
does this work, after the process has exited?

Reply:
Yes, in my tests the diagnostic info (_RSS_, _native memory tracking_, ...) were successfully retrieved from this `InputStream`.